app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypes.kt

```diff
@@ -155,23 +169,51 @@ internal fun MessageAsset(assetName: String, assetExtension: String, assetSizeIn
                     color = MaterialTheme.wireColorScheme.secondaryText,
                     style = MaterialTheme.wireTypography.subline01
                 )
-                Text(
-                    modifier = Modifier
-                        .padding(start = MaterialTheme.wireDimensions.spacing8x)
-                        .constrainAs(downloadStatus) {
-                            top.linkTo(parent.top)
-                            end.linkTo(parent.end)
-                            bottom.linkTo(parent.bottom)
+                Row(
+                    modifier = Modifier.wrapContentWidth().constrainAs(downloadStatus) {
+                        top.linkTo(parent.top)
+                        end.linkTo(parent.end)
+                        bottom.linkTo(parent.bottom)
+                    },
+                ) {
+                    Text(
+                        modifier = Modifier.padding(end = dimensions().spacing4x),
+                        text = getDownloadStatusText(assetDownloadStatus),
+                        color = MaterialTheme.wireColorScheme.run {
+                            if (assetDownloadStatus == FAILED) error else secondaryText
                         },
-                    text = stringResource(R.string.asset_message_download_text),
-                    color = MaterialTheme.wireColorScheme.secondaryText,
-                    style = MaterialTheme.wireTypography.subline01
-                )
+                        style = MaterialTheme.wireTypography.subline01
+                    )
+
+                    // Extra download status icon
+                    when (assetDownloadStatus) {
+                        IN_PROGRESS -> WireCircularProgressIndicator(
+                            progressColor = MaterialTheme.wireColorScheme.secondaryText,
+                            size = dimensions().spacing16x
+                        )
+                        DOWNLOADED -> Icon(
+                            painter = painterResource(id = R.drawable.ic_check_tick),
+                            contentDescription = stringResource(R.string.content_description_check),
+                            modifier = Modifier.size(dimensions().wireIconButtonSize),
+                            tint = MaterialTheme.wireColorScheme.secondaryText
+                        )
```

Comment:
instead of the comment I would move it to a private function with a perfect name mentioned in the comment :P